### PR TITLE
Prime factorization example

### PR DESCRIPTION
**Pull request by morganthomas**

### CODE REVIEW

File: prime_factorization/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        match valida_rs::io::read_line::<u32>() {
+            Ok(num) => break num,
+            Err(e) => {
+                valida_rs::io::println(&format!("Error reading input: {}. Please try again:", e));
+            }
+        }
+    }
+}
+
+pub fn check_prime_factorization(x: u32, ys: Vec<u32>) -> bool {
```
Comment (Avi-D-coder) [inline prime_factorization/src/lib.rs:12]: NIT: `&[u32]` or `impl AsRef<[u32]>` is more idiomatic.

File: prime_factorization/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Fibonacci: A template project for Valida
+
+This is a simple program that calculates the n-th fibonacci number and proves the execution of the computation in Valida. You can use this as a template for your projects which create Valida proofs of execution of Rust code.
+
```
Comment (Avi-D-coder) [inline prime_factorization/README.md:4]: This is out of date

File: prime_factorization/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    let mut z = 1;
+    for y in ys {
+        z *= y;
+    }
+    z == x
```
Comment (Avi-D-coder) [inline prime_factorization/src/lib.rs:17]: This is just a factorization check, we are not checking if each y is prime.

File: prime_factorization/LICENSE [hunk 4]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (thealmarty) [inline prime_factorization/LICENSE:1]: do we need another license for each crate? There's one in the main directory.

File: prime_factorization/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+#![no_main]
+
+use prime_factorization::{read_number, check_prime_factorization};
+
+valida_rs::entrypoint!(main);
+
+pub fn main() {
+    valida_rs::io::println("Please enter a 32-bit number:");
+    // Read a line from stdin and parse it as an u8.
```
Comment (thealmarty) [inline prime_factorization/src/main.rs:9]: u32 instead of u8

**Comment by morganthomas** [comment]
Thanks for the feedback, Marty and Avi! I addressed all of it.

**Pull request merged**